### PR TITLE
VxWorks/C++ fixes

### PR DESCRIPTION
**Pull request by timcanham**
| | |
|:---|:---|
|**_Originating Project/Creator_**|@timcanham |
|**_Affected Component_**| None, PrmBuffer.hpp/.cpp |
|**_Affected Architectures(s)_**| All, portable code |
|**_Related Issue(s)_**| None |
|**_Has Unit Tests (y/n)_**| No, but verfied PrmDb |
|**_Builds Without Errors (y/n)_**| y |
|**_Unit Tests Pass (y/n)_**| y, PrmDb |
|**_Documentation Included (y/n)_**| n/a |

---
## Change Description

Updated `Fw/PrmBuffer` private data member from `m_data` to `m_bufferData` to avoid collision with VxWorks `m_data` macro.

## Rationale

Needs to be fixed to support VxWorks builds

## Testing/Review Recommendations

Re-ran `PrmDb` as a check, but there are no references to the `m_data` member in `PrmDb` or the `PrmDb` unit tests.

## Future Work

Note any additional work that will be done relating to this issue.

### CODE REVIEW

File: Svc/ActiveTextLogger/ActiveTextLoggerImpl.cpp [hunk 1]
```diff
@@ -130,7 +131,7 @@ namespace Svc {
     {
 
         // Print to console:
-        (void) printf("%s",text.toChar());
```
Comment (LeStarch) [inline Svc/ActiveTextLogger/ActiveTextLoggerImpl.cpp:133]: THANK YOU!!

File: Svc/BufferManager/BufferManagerComponentImpl.cpp [hunk 2]
```diff
@@ -203,8 +203,11 @@ namespace Svc {
         }
     }
       
-    // check some assertions
-    FW_ASSERT(bufferMem == (static_cast<U8*>(memory) + memorySize));
+    // check that the initiation pointer made it to the end of allocated space
+    U8* const CURR_PTR = bufferMem;
+    U8* const END_PTR = static_cast<U8*>(memory) + memorySize;
+    FW_ASSERT(CURR_PTR == END_PTR, reinterpret_cast<U64>(CURR_PTR), reinterpret_cast<U64>(END_PTR));
```
Comment (LeStarch) [inline Svc/BufferManager/BufferManagerComponentImpl.cpp:209]: These U64's should really be POINTER_CAST as U64 pointers aren't supported on smaller platforms.

**Review by LeStarch — CHANGES_REQUESTED** [review]
Just one change, used POINTER_CAST as the casting type for pointer -> int.

File: Svc/BufferManager/BufferManagerComponentImpl.cpp (hunk 2 above)
Comment (timcanham) [reply Svc/BufferManager/BufferManagerComponentImpl.cpp]: Change is made...

**Review by LeStarch — APPROVED** [review]

**Pull request merged**